src/main.rs

```diff
@@ -66,10 +70,31 @@ fn write_err(err: StratisError) -> StratisResult<()> {
 }
 
 fn main() {
-
-    let engine = Rc::new(RefCell::new(SimEngine::new()));
-    // TODO: add cmdline option to specify engine
-    //  let context = Rc::new(RefCell::new(Context::new()));
+    let matches = App::new("stratis")
+        .version(&crate_version!())
+        .about("Stratis storage management")
+        .arg(Arg::with_name("debug")
+            .short("d")
+            .long("debug")
+            .help("Print additional output for debugging"))
+        .arg(Arg::with_name("sim")
+            .long("sim")
+            .help("Use simulator engine"))
+        .get_matches();
+
+    if matches.is_present("debug") {
+        unsafe { debug = true }
+    };
+
+    let engine: Box<Engine> = {
```

Comment:
Can you open an issue with your current thinking, so we can start discussing the level of flexibility we need? Does "programmatically accessible" mean runtime engine switching? Not sure we need that.